lib/src/screens/setup_2fa/setup_2fa_enter_code_page.dart

```diff
@@ -0,0 +1,228 @@
+    totpController.addListener(() {
+      setup2FAViewModel.enteredOTPCode = totpController.text;
+    });
+  }
+
+  @override
+  String get title => isForSetup ? S.current.setup_2fa : S.current.verify_with_2fa;
+
+  Widget? leading(BuildContext context) {
+    return closable ? super.leading(context) : null;
+  }
+
+  final TextEditingController totpController;
+  final Setup2FAViewModel setup2FAViewModel;
+  final bool isForSetup;
+  final bool closable;
+
+  @override
+  Widget body(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(
+        horizontal: 24,
+      ),
+      child: Column(
+        children: [
+          BaseTextFormField(
+            textAlign: TextAlign.left,
+            hintText: S.current.totp_code,
+            controller: totpController,
+            keyboardType: TextInputType.number,
+            textStyle: TextStyle(fontSize: 16, fontWeight: FontWeight.w600, color: Colors.black),
+            placeholderTextStyle: TextStyle(
+              fontSize: 16,
+              fontWeight: FontWeight.w600,
+            ),
+          ),
+          SizedBox(height: 16),
+          Text(
+            S.current.please_fill_totp,
+            style: TextStyle(
+              fontSize: 12,
+              fontWeight: FontWeight.w400,
+              height: 1.2,
+              color: Palette.darkGray,
+            ),
+            textAlign: TextAlign.center,
+          ),
+          Spacer(),
+          Observer(
+            builder: (context) {
+              return PrimaryButton(
+                isDisabled: setup2FAViewModel.enteredOTPCode.length != 8,
+                onPressed: () async {
+                  final result =
+                      await setup2FAViewModel.totp2FAAuth(totpController.text, isForSetup);
+                  final bannedState = setup2FAViewModel.state is AuthenticationBanned;
+
+                  await showPopUp<void>(
+                    context: context,
+                    builder: (BuildContext context) {
+                      return PopUpCancellableAlertDialog(
+                        contentText: () {
+                          switch (result) {
+                            case true:
+                              return isForSetup
+                                  ? S.current.totp_2fa_success
+                                  : S.current.totp_verification_success;
+                            case false:
+                              if (bannedState) {
+                                final state = setup2FAViewModel.state as AuthenticationBanned;
+                                return S.of(context).failed_authentication(state.error);
+                              } else {
+                                return S.current.totp_2fa_failure;
+                              }
+                            default:
+                              return S.current.enter_totp_code;
+                          }
+                        }(),
```

Comment:
I'll extract it into a function.

The reason for the switch is to keep it easier to visualise as we're running other checks if the result is either true or false.

Reply:
since it's already a function you can directly call it instead of wrapping it in a function body and call it
i.e instead of this
```Dart
() {
 // ...
}()
```